src/components/pages/maintenance/ruleDetails/ruleDetails.js

```diff
@@ -1,12 +1,248 @@
+      paginationPageSize: 8,
+      t
+    };
+
+    const { selectedTab, selectedAlert = {} } = this.state;
+    const { counts = {} } = selectedAlert;
+    return [
+      <ContextMenu key="context-menu">
+        <RefreshBar
+          refresh={refreshData}
+          time={lastUpdated}
+          isPending={isPending}
+          t={t} />
+      </ContextMenu>,
+      <PageContent className="maintenance-container rule-details-container" key="page-content">
+        <div className="header-container">
+          <h1 className="maintenance-header">{alarmName}</h1>
+          <div className="rule-stat-container">
+            <div className="rule-stat-cell">
+              <div className="rule-stat-header">Total</div>
+              <div className="rule-stat-value">{renderCount(counts.total)}</div>
+            </div>
+            <div className="rule-stat-cell">
+              <div className="rule-stat-header">Open</div>
+              <div className="rule-stat-value">{renderCount(counts.open)}</div>
+            </div>
+            <div className="rule-stat-cell">
+              <div className="rule-stat-header">Ack</div>
+              <div className="rule-stat-value">{renderCount(counts.acknowledged)}</div>
+            </div>
+            <div className="rule-stat-cell">
+              <div className="rule-stat-header">Closed</div>
+              <div className="rule-stat-value">{renderCount(counts.closed)}</div>
+            </div>
+            <div className="rule-stat-cell">
+              <div className="rule-stat-header">Last Event</div>
+              <div className="rule-stat-value">
+                {
+                  selectedAlert.lastOccurrence
+                    ?  <LastTriggerRenderer context={({ t: this.props.t })} value={selectedAlert.lastOccurrence} />
+                    : '--'
+                }
+              </div>
+            </div>
+            <div className="rule-stat-cell">
+              <div className="rule-stat-header">Severity</div>
+              <div className="rule-stat-value">
+                {
+                  selectedAlert.severity
+                    ?  <SeverityRenderer context={({ t: this.props.t })} value={selectedAlert.severity} />
+                    : '--'
+                }
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="details-description">
+          Manage alarm occurences associated to this rule in the section below, and use the associated information to troubleshoot each occurence.
+        </div>
+        <h4 className="sub-heading">Rule detail</h4>
+        <RulesGrid rowData={rule} t={t} pagination={false} />
+
+        <h4 className="sub-heading">Alarm occurrences</h4>
+        <AlarmOccurrencesGrid {...alarmsGridProps} />
+
+        <h4 className="sub-heading">Related information</h4>
+        <div className="tab-container">
+          <button className={joinClasses('tab', selectedTab === 'all' ? 'active' : '')}
```

Comment:
constants for the magical strings? Or consider doing the comparisons and saving into a local variable so the comparison doesn't have to be done multiple times (for example, selectedTab === 'devices' comparison is done on line 224 & 230).